rhombus/private/string.rkt

```diff
@@ -56,7 +56,8 @@
 (define/arity (to_string a)
   #:static-infos ((#%call-result #,string-static-infos))
   (cond
-    [(string? a) a]
+    [(immutable-string? a) a]
```

Comment:
`string->immutable-string` already has this optimization, there's no need to do it here too.